### PR TITLE
Add a command palette that should show up on Cmd/Ctrl+K

### LINKED ISSUE #4617: [UX] Command Palette
[Figma link](https://www.figma.com/file/SSr1ljyzF0lCXTL7BX1Whh/?node-id=8958%3A84248)

### PR DESCRIPTION
**Pull request by jaril**
Fix #4617.

Remaining things to finish here before merging V1:
- [x] Add keyboard shortcuts for each of the commands **(edit: skipped for now since we need to come up with proper ones)**
- [x] Let the user trigger a keyboard shortcut while focused on an input, e.g. full text search input prevents keyboard shortcuts 
- [x] Do a design pass with Jon

### CODE REVIEW

File: src/ui/reducers/layout.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      };
+    }
+
+    default: {
+      return state;
+    }
+  }
+}
+
+export const getShowCommandPalette = (state: UIState) => state.layout.showCommandPalette;
```
Comment (jcmorrow) [inline src/ui/reducers/layout.ts:26]: It's a shame, this is *almost* simple enough that it doesn't belong in redux, but it's probably right to have it here.

**Review by jcmorrow — COMMENTED** [review]
This all looks pretty reasonable, I left a few comments on style in various places. When testing locally I got this error:

CleanShot 2021-12-08 at 16 04 18

I'm having trouble reproducing it for a Replay now. Also one note: it feels bad that you can't arrow up and down through the menu, would that be an easy add?

File: src/devtools/client/debugger/src/components/FullTextSearch/FullTextFilter.js [hunk 2]
```diff
@@ -12,8 +12,6 @@ export function FullTextFilter({ results, onKeyDown }) {
       return;
     }
 
-    e.stopPropagation();
```
[the remark this answers is not in the document]
Comment (jaril) [reply src/devtools/client/debugger/src/components/FullTextSearch/FullTextFilter.js:15]: This was keeping the keyboard shortcuts from being triggered when the user was focused inside the input box.

File: src/ui/components/CommandPalette/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import CommandPalette from "./CommandPalette";
+
+export default CommandPalette;
```
[the remark this answers is not in the document]
Comment (jaril) [reply src/ui/components/CommandPalette/index.ts:3]: Yeah, just got irked at having to Cmd+P then open `index.tsx` instead of just `CommandPalette.tsx`.

**Pull request merged**